### PR TITLE
chore: run bulk action interstitial component

### PR DESCRIPTION
**Pull request by ashtonG**
## Ticket
[ET-94]

## Description
this adds an interstitial component for run bulk actions. when opened, it runs a search given the current selection and filters and determines whether or not the selection contains any runs that are part of a search, returning the results to the opener. as a side effect, we pull out the `mergeAbortControllers` helper from `trialDetailsLogs` while also adding test coverage to that helper.

## Test Plan
Component is not currently user-facing. Unit tests should comprehensively check functionality.

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

[ET-94]: https://hpe-aiatscale.atlassian.net/browse/ET-94?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/src/components/RunFilterInterstitialModalComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+          const results = await searchRuns(
+            {
+              filter: JSON.stringify(filter),
+              limit: 0,
+              projectId,
+            },
+            { signal: mergedCanceler.signal },
+          );
+
+          return (results.pagination.total || 0) > 0;
```
Comment (keita-determined) [inline webui/react/src/components/RunFilterInterstitialModalComponent.tsx:116]: this should be wrapped by `Loaded`?

File: webui/react/src/components/RunFilterInterstitialModalComponent.tsx
Comment (keita-determined) [inline webui/react/src/components/RunFilterInterstitialModalComponent.tsx:1]: how can i use this modal? i tried it on local, but i couldn't trigger the modal open (my settings might be wrong).

**Review by keita-determined — COMMENTED** [review]
i wasnt sure how to use the modal on local, i'll reach out to you tomorrow

File: webui/react/src/components/RunFilterInterstitialModalComponent.tsx (hunk 1 above)
Comment (ashtonG) [reply webui/react/src/components/RunFilterInterstitialModalComponent.tsx:135]: useAsync wraps the value for us. we only need to return a loadable when we want to force the value to return `NotLoaded` or `Failed`.

**Review by keita-determined — APPROVED** [review]
LGTM
the usage of this is 
- call `RunFilterInterstitialModalComponent` first
- get the close reason
- if statement
  - show `RunMoveWarningModalComponent` when `has_search_runs` is returned
  - show move modal when `no_search_runs` is returned without showing `RunMoveWarningModalComponent`
  - show error message when `failed` is retuend

is that correct?

**Comment by ashtonG** [comment]
i think we should show the move modal first -- that way we defer the check until after the user confirms the details of the move.

**Comment by keita-determined** [comment]
got it. if its ready to get merged, could you merge it to work on the move action?

**Pull request merged**